central/complianceoperator/v2/report/manager/manager_impl.go

```diff
@@ -337,7 +351,42 @@ func (m *managerImpl) HandleScan(sensorCtx context.Context, scan *storage.Compli
 		}
 		return err
 	}
-	return m.getWatcher(sensorCtx, id).PushScan(scan)
+	log.Debugf("Searching for ScanWatcher %s to PushScan", id)
+	numChecks, err := watcher.GetExpectedNumChecks(scan)
+	if err != nil {
+		log.Warnf("Failed to get expected number of checks from annotations for %s: %v", scan.GetScanName(), err)
+	}
+	w := m.getWatcher(sensorCtx, id, numChecks)
+	if w != nil {
+		return w.PushScan(scan)
+	}
+	return nil
```

Comment:
@lvalerom shall we return an error here or better stay with nil?

Reply:
We should not return an error because technically this is not an error, we just received a scan update after the watcher was done (which is ok). I'd maybe log something here (debug level) just in case.

Reply:
Added those in [e7e9ef5](https://github.com/stackrox/stackrox/pull/15517/commits/e7e9ef586693054bdaf01dc34c35099aa558ffb8)